scene/gui/texture_progress_bar.cpp

```diff
@@ -428,36 +428,18 @@ void TextureProgressBar::draw_nine_patch_stretched(const Ref<Texture2D> &p_textu
 void TextureProgressBar::_notification(int p_what) {
 	switch (p_what) {
 		case NOTIFICATION_DRAW: {
-			if (nine_patch_stretch && (mode == FILL_LEFT_TO_RIGHT || mode == FILL_RIGHT_TO_LEFT || mode == FILL_TOP_TO_BOTTOM || mode == FILL_BOTTOM_TO_TOP || mode == FILL_BILINEAR_LEFT_AND_RIGHT || mode == FILL_BILINEAR_TOP_AND_BOTTOM)) {
-				if (under.is_valid()) {
+			if (under.is_valid()) {
+				if (nine_patch_stretch) {
 					draw_nine_patch_stretched(under, mode, 1.0, tint_under);
+				} else {
+					draw_texture(under, Point2(), tint_under);
 				}
-				if (progress.is_valid()) {
+			}
+
+			if (progress.is_valid()) {
+				if (nine_patch_stretch && (mode == FILL_LEFT_TO_RIGHT || mode == FILL_RIGHT_TO_LEFT || mode == FILL_TOP_TO_BOTTOM || mode == FILL_BOTTOM_TO_TOP || mode == FILL_BILINEAR_LEFT_AND_RIGHT || mode == FILL_BILINEAR_TOP_AND_BOTTOM)) {
```

Comment:
I'd add a bool called `is_radial_mode`, which would be defined by the same condition as the one below (checking for radial modes). Then this big condition can be replaced with `if (nine_patch_stretch && !is_radial_mode)`.

Reply:
Done!